### PR TITLE
Bflow vtkm

### PR DESCRIPTION
**Pull request by sshudler**
Merging an example for a BabelFlow-based extract that shows how to compute iso surfaces with BabelFlow

### CODE REVIEW

**Comment by cyrush** [comment]
@sshudler -- thanks! we have a few things in flux related to vtk-m updates. Those need to get worked out so we can update the CI containers. I'll circle back when those are resolved.

**Comment by cyrush** [comment]
@sshudler, babelflow is back up and testing in CI as of #691. Thanks for your help working through those issues.

Q: Was the work in this PR also included in those branch updates, or do we also need to work to get this merged?

**Comment by sshudler** [comment]
@cyrush We need to do some work to get this branch merged as well. The other branch had only updates related to tags in BabelFlow and PMT. I'll go ahead and fix the conflict in `src/ascent/CMakeLists.txt` - it relates to one of the BabelFlow extracts

**Comment by sshudler** [comment]
How do I resolve the conflict in `azure-pipelines.yml`?

Is it the first or the second option below?

bflow-vtkm
  main_tag : alpinedav/ascent-ci:ubuntu-18-devel-tpls_2021-01-25-sha3769e1
  cuda_tag : alpinedav/ascent-ci:ubuntu-16-cuda-10.1-devel-tpls_2021-01-25-sha3769e1

  main_tag : alpinedav/ascent-ci:ubuntu-18-devel-tpls_2021-03-18-shae218ec
  cuda_tag : alpinedav/ascent-ci:ubuntu-18-cuda-10.1-devel-tpls_2021-03-18-shae218ec
develop

**Comment by cyrush** [comment]
You want  the second (newer containers) that are currently on develop:

To shine a little be of light on the naming, the date is included as part of the container:

main_tag : alpinedav/ascent-ci:ubuntu-18-devel-tpls_**2021-03-18**-shae218ec
cuda_tag : alpinedav/ascent-ci:ubuntu-18-cuda-10.1-devel-tpls_**2021-03-18**-shae218ec

These containers have the tagged versions of babelflow and pmt

**Comment by sshudler** [comment]
I resolved the conflicts

File: src/ascent/CMakeLists.txt [hunk 1]
```diff
@@ -369,8 +369,7 @@ if(MPI_FOUND)
       list(APPEND ascent_sources runtimes/flow_filters/ascent_runtime_babelflow_pmt.cpp)
       list(APPEND ascent_sources runtimes/flow_filters/ascent_runtime_babelflow_comp_utils.hpp)
       list(APPEND ascent_sources runtimes/flow_filters/ascent_runtime_babelflow_comp_utils.cpp)
-      list(APPEND ascent_sources runtimes/flow_filters/ascent_runtime_babelflow_compose.cpp)
-      list(APPEND ascent_sources runtimes/flow_filters/ascent_runtime_babelflow_iso.cpp)
+      list(APPEND ascent_sources runtimes/flow_filters/ascent_runtime_babelflow_compose.cpp)
```
Comment (cyrush) [inline src/ascent/CMakeLists.txt:372]: @sshudler just want to double check, are we ready to merge? It seems to only remove one source file from the CMakeLists.txt. (I assume the other changes were present in other another branch?)

Comment (sshudler) [reply src/ascent/CMakeLists.txt:372]: @cyrush It's strange - it shows 10 commits, but only 1 file changed. Then the last commit in the `bflow-vtkm` branch is "Merge branch 'develop' into bflow-vtkm" by me... I'm not sure - did it happens when I fixed the two conflicts? I don't understand what happend

Comment (sshudler) [reply src/ascent/CMakeLists.txt:372]: Looks like there is a commit in `develop` called "2020 12 update vtkm (#678)" that includes some (or probably) most of the commits from this branch. I'm not sure how to explain that, but I guess this is why the PR only shows one file has changed

**Comment by cyrush** [comment]
@sshudler not sure what happened, but I think the best course would be to start a new branch off of develop and get your changes in that way.

File: src/ascent/CMakeLists.txt (hunk 1 above)
Comment (cyrush) [reply src/ascent/CMakeLists.txt:372]: ahh, so does that mean we have the bulk of changes in? If so --  are we good to merge this to finish the final changes needed?

Comment (sshudler) [reply src/ascent/CMakeLists.txt:372]: yes, it appears most of the changes are in and we're good to go with the merge, which will be only this single file

Comment (cyrush) [reply src/ascent/CMakeLists.txt:372]: great, thanks for clarifying!

**Pull request merged**